src/main/java/com/google/planningworks/cdap/plugins/source/AnaplanExportConfig.java

```diff
@@ -0,0 +1,506 @@
+  public static final String NAME_SERVICE_ACCOUNT_JSON = "serviceAccountJSON";
+  public static final String NAME_PATH = "path";
+  public static final String NAME_FILE_SYSTEM_PROPERTIES = "fileSystemProperties";
+  public static final String NAME_FILE_REGEX = "fileRegex";
+  public static final String NAME_FORMAT = "format";
+  public static final String NAME_DELIMITER = "delimiter";
+  public static final String SERVICE_ACCOUNT_FILE_PATH = "filePath";
+  public static final String SERVICE_ACCOUNT_JSON = "JSON";
+
+  public static final String NAME_ARCHIVE_NAME = "archiveName";
+  public static final String NAME_BUCKET = "bucket";
+
+  private static final String DEFAULT_ENCRYPTED_METADATA_SUFFIX = ".metadata";
+
+  private static final Gson GSON = new Gson();
+  private static final Type MAP_STRING_STRING_TYPE =
+    new TypeToken<Map<String, String>>() {
+    }.getType();
+
+  @Name(Constants.Reference.REFERENCE_NAME)
+  @Description(
+    "This will be used to uniquely identify this source for lineage, annotating metadata, etc.")
+  public String referenceName;
+
+  // GCP service account properties
+  @Name(NAME_PROJECT)
+  @Description(
+    "Google Cloud Project ID, which uniquely identifies a project. "
+      + "It can be found on the Dashboard in the Google Cloud Platform Console.")
+  @Macro
+  @Nullable
+  protected String project;
+
+  @Name(NAME_SERVICE_ACCOUNT_TYPE)
+  @Description(
+    "Service account type, file path where the service account is located or the JSON content of"
+      + " the service account.")
+  @Macro
+  @Nullable
+  protected String serviceAccountType;
+
+  @Name(NAME_SERVICE_ACCOUNT_FILE_PATH)
+  @Description(
+    "Path on the local file system of the service account key used for authorization. Can be set"
+      + " to 'auto-detect' when running on a Dataproc cluster. When running on other clusters,"
+      + " the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  protected String serviceFilePath;
+
+  @Name(NAME_SERVICE_ACCOUNT_JSON)
+  @Description("Content of the service account file.")
+  @Macro
+  @Nullable
+  protected String serviceAccountJson;
+
+  // Anaplan properties
+  @Name(AnaplanService.NAME_SERVER_FILE_NAME)
+  @Macro
+  @Description("Anaplan Server file name")
+  private String serverFileName;
+
+  @Name(NAME_ARCHIVE_NAME)
+  @Macro
+  @Description("File archive name in GCS bucket")
+  private String archiveName;
+
+  // GCS properties
+  @Name(NAME_BUCKET)
+  @Macro
+  @Description("GCS bucket for export file buffer")
+  private String bucket;
+
+  @Macro
+  @Nullable
+  @Description("Map of properties to set on the InputFormat.")
+  private String fileSystemProperties;
+
+  @Macro
+  @Nullable
+  @Description(
+    "Maximum size of each partition used to read data. Smaller partitions will increase the level"
+      + " of parallelism, but will require more resources and overhead.")
+  private Long maxSplitSize;
```

Comment:
max and min split size is used for fairly low level performance tuning in the normal file source. Since this plugin does a serial copy from anaplan to GCS, I don't think we'll be dealing with data volumes where this level of tuning needs to be exposed. Let's remove both of them.

Reply:
Removed